central/scannerv4definitions/handler/handler.go

```diff
@@ -0,0 +1,275 @@
+package handler
+
+import (
+	"archive/zip"
+	"context"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	timestamp "github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	blob "github.com/stackrox/rox/central/blob/datastore"
+	"github.com/stackrox/rox/central/blob/snapshot"
+	"github.com/stackrox/rox/central/scannerdefinitions/file"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
+)
+
+const (
+	baseDir = "scannerv4mappings"
+
+	// mappingFileName represents the offline zip bundle.
+	mappingFileName = "mapping-data.zip"
+
+	// mappingfileBlobName represents the blob name of offline/fallback zip bundle for repo mapping data for Scanner.
+	mappingfileBlobName = "/offline/scannerV4/" + mappingFileName
+
+	repoMappingURL = "https://storage.googleapis.com/scanner-v4-test/redhat-repository-mappings/"
+
+	defaultUpdateInterval = 4 * time.Hour
+
+	name2cpe = "container-name-repos-map.json"
+
+	repo2cpe = "repository-to-cpe.json"
+)
+
+var (
+	client = &http.Client{
+		Transport: proxy.RoundTripper(),
+		Timeout:   5 * time.Minute,
+	}
+
+	log     = logging.LoggerForModule()
+	randGen = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	fileMapping = map[string]string{
+		"name2cpe": name2cpe,
+		"repo2cpe": repo2cpe,
+	}
+)
+
+type requestedUpdater struct {
+	*mappingUpdater
+	lastRequestedTime time.Time
+}
+
+// httpHandler handles HTTP GET and POST requests for vulnerability data.
+type httpHandler struct {
+	interval  time.Duration
+	lock      sync.Mutex
+	updater   *requestedUpdater
+	dataDir   string
+	blobStore blob.Datastore
+}
+
+func (h *httpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case http.MethodGet:
+		h.get(w, r)
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
+
+// New creates a new http.Handler to handle repo mapping data.
+func New(blobStore blob.Datastore) http.Handler {
+	h := &httpHandler{
+		interval:  env.RepoMappingUpdateMaxInitialWait.DurationSetting(),
+		blobStore: blobStore,
+	}
+	h.initializeUpdater(context.Background())
+	return h
+}
+
+func (h *httpHandler) initializeUpdater(ctx context.Context) {
+	var err error
+	utils.CrashOnError(err)
+
+	h.updater = &requestedUpdater{}
+	go h.fetchMappingData(ctx)
+}
+
+func (h *httpHandler) fetchMappingData(ctx context.Context) {
+	ticker := time.NewTicker(env.RepoMappingUpdateMaxInitialWait.DurationSetting())
+	defer ticker.Stop()
+	log.Infof("Starting the updater loop")
```

Comment:
This is pretty generic. Updating what?

Also, just `log.Info` instead of `log.Infof`